clang/test/SemaSYCL/sycl-dllimport-dllexport.cpp

```diff
@@ -0,0 +1,48 @@
+// RUN: %clang_cc1 -triple spir64-unknown-linux-sycldevice -fsycl-is-device -fms-extensions -DTEST %s -o /dev/null 2>&1 | FileCheck %s --check-prefixes CHECK
+
+// RUN: %clang_cc1 -triple spir64-unknown-windows-sycldevice -fms-extensions -DTEST %s -o /dev/null 2>&1 | FileCheck %s --check-prefixes CHECK
+
+// RUN: %clang_cc1 -triple spir64-unknown-windows-sycldevice -fsycl-is-device -fms-extensions -DTEST %s -o /dev/null 2>&1 | FileCheck %s --check-prefixes CHECKALL
+
+// RUN: %clang_cc1 -triple spir64-unknown-windows-sycldevice -fsycl-is-device -fms-extensions -fsyntax-only -verify  %s
+
+#if defined(TEST)
+// CHECK: warning: __declspec attribute 'dllexport' is not supported
+int __declspec(dllexport) foo(int a) {
+  return a;
+}
+
+// CHECK: warning: __declspec attribute 'dllimport' is not supported
+int __declspec(dllimport) bar();
+
+// CHECK: warning: unknown attribute 'dllimport' ignored
+int [[dllimport]]xoo();
+
+// CHECKALL: warning: unknown attribute 'dllimport' ignored
+int zoo() __attribute__((dllimport));
+
+#else
+
+// emit error if dllimport function is called in sycl kernel
+int  __declspec(dllexport) foo(int a) {
+   return a;
+}
+// expected-note@+1 {{'bar' declared here}}
+int __declspec(dllimport) bar();
+
+template <typename name, typename Func>
+__attribute__((sycl_kernel)) void kernel_single_task(Func kernelFunc) {
+  kernelFunc();
+}
+
+int main() {
+  foo(10);  // expected-no-error
+  bar();  // expected-no-error
+  kernel_single_task<class fake_kernel>([]() {
+    foo(10);// expected-no-error
+    bar(); // expected-error {{SYCL kernel cannot call a dllimport function}}
```

Comment:
Please, add following test case:
```c++
int __declspec(dllimport) bar();
int baz() { return bar(); }
// call baz from a SYCL kernel.

Reply:
Why you want do define a dllimport function.  Definition of dllimport function not allowed.

Reply:
I think the idea is that it gives an error if called?

Reply:
It to me call to dllexport function should be OK.  With your idea, we should add LangOpt for SYCL for it right.

Reply:
@bader and @erichkeane :  I add test for this, which we don't expect and error when call inside sycl kernel, since dllimport gets ignored.

Thanks.